packages/astro/src/core/add/index.ts

```diff
@@ -0,0 +1,417 @@
+				}
+				if (prop.key.type === 'StringLiteral') {
+					if (prop.key.value === 'integrations') return true;
+				}
+				return false;
+			}) as t.ObjectProperty | undefined;
+
+			const integrationCall = t.callExpression(integrationId, []);
+
+			if (!integrationsProp) {
+				configObject.properties.push(t.objectProperty(t.identifier('integrations'), t.arrayExpression([integrationCall])));
+				return;
+			}
+
+			if (integrationsProp.value.type !== 'ArrayExpression') throw new Error('Unable to parse integrations');
+
+			const existingIntegrationCall = integrationsProp.value.elements.find(
+				(expr) => t.isCallExpression(expr) && t.isIdentifier(expr.callee) && expr.callee.name === integrationId.name
+			);
+
+			if (existingIntegrationCall) return;
+
+			integrationsProp.value.elements.push(integrationCall);
+		},
+	});
+}
+
+const enum UpdateResult {
+	none,
+	updated,
+	cancelled,
+	failure,
+}
+
+async function updateAstroConfig({ configURL, ast, logging }: { logging: LogOptions; configURL: URL; ast: t.File }): Promise<UpdateResult> {
+	const input = await fs.readFile(fileURLToPath(configURL), { encoding: 'utf-8' });
+	let output = await generate(ast);
+	const comment = '// https://astro.build/config';
+	const defaultExport = 'export default defineConfig';
+	output = output.replace(` ${comment}`, '');
+	output = output.replace(`${defaultExport}`, `\n${comment}\n${defaultExport}`);
+
+	if (input === output) {
+		return UpdateResult.none;
+	}
+
+	let changes = [];
+	for (const change of diffWords(input, output)) {
+		let lines = change.value.trim().split('\n').slice(0, change.count).filter(x => x && !INSIGNIFICANT_CHARS.has(x));
+		if (lines.length === 0) continue;
+		if (change.added) {
+			if (INSIGNIFICANT_CHARS.has(change.value.trim())) continue;
+			changes.push(change.value);
+		}
+	}
+	let diffed = output;
+	for (let newContent of changes) {
+		const coloredOutput = newContent.split('\n').map(ln => ln ? green(ln) : '').join('\n');
+		diffed = diffed.replace(newContent, coloredOutput);
+	}
+
+	const message = `\n${boxen(
+		diffed,
+		{ margin: 0.5, padding: 0.5, borderStyle: 'round', title: configURL.pathname.split('/').pop() }
+	)}\n`;
+
+	info(logging, null, `\n  ${magenta('Astro will update your configuration with these changes...')}\n${message}`);
+
+	const response = await prompts({
+		type: 'confirm',
+		name: 'updateConfig',
+		message: 'Continue?',
+		initial: true,
+	});
+
+	if (response.updateConfig) {
+		await fs.writeFile(fileURLToPath(configURL), output, { encoding: 'utf-8' });
+		debug('add', `Updated astro config`);
+		return UpdateResult.updated;
+	} else {
+		return UpdateResult.cancelled;
+	}
+}
+
+interface InstallCommand {
+	pm: string;
+	command: string;
+	flags: string[];
+	dependencies: string;
+}
+async function getInstallIntegrationsCommand({ integrations, cwd = process.cwd() }: { integrations: IntegrationInfo[]; cwd?: string }): Promise<InstallCommand | null> {
```

Comment:
Why do you return an object instead of a string?

Reply:
Just to improve highlighting the message! Not super important, but a nice touch.